### PR TITLE
`sage.{geometry,rings}`: More `# optional` tags in doctests

### PR DESCRIPTION
**Pull request by mkoeppe**
### 📚 Description

Adding more `# optional` tags in doctests so that doctests can be run with modularized distributions.

### 📝 Checklist

- [x] I have made sure that the title is self-explanatory and the description concisely explains the PR.
- [ ] I have linked an issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by kliem — APPROVED** [review]

File: src/sage/geometry/polyhedron/base2.py [hunk 1]
```diff
@@ -784,16 +784,16 @@ def generating_function_of_integral_points(self, **kwds):
...
-            sage: P.<z> = PowerSeriesRing(ZZ)
-            sage: c = f.subs({y: z for y in f.parent().gens()}); c
+            sage: f = f.value()                                                                 # optional - sage.combinat
+            sage: P.<z> = PowerSeriesRing(ZZ)                                                   # optional - sage.combinat
+            sage: c = f.subs({y: z for y in f.parent().gens()}); c                              # optional - sage.combinat
             z^5 + z^6 + 2*z^7 + 3*z^8 + 5*z^9 + 7*z^10 + 10*z^11 + 13*z^12 + 18*z^13 +
             23*z^14 + 30*z^15 + 37*z^16 + 47*z^17 + 57*z^18 + 70*z^19 + 84*z^20 +
             101*z^21 + 119*z^22 + 141*z^23 + 164*z^24 + O(z^25)
-            sage: [Partitions(k, length=5).cardinality() for k in range(5,20)] == \
+            sage: [Partitions(k, length=5).cardinality() for k in range(5,20)] == \             # optional - sage.combinat
```
Comment (kliem) [inline src/sage/geometry/polyhedron/base2.py]: The `# optional` needs to be in the second line here.

File: src/sage/rings/integer.pyx [hunk 2]
```diff
@@ -498,7 +498,7 @@ cdef class Integer(sage.structure.element.EuclideanDomainElement):
 
         ::
 
-            sage: k = GF(2)
+            sage: k = GF(2)                                 # optional - sage.libs.pari
             sage: ZZ( (k(0),k(1)), 2)
```
Comment (kliem) [inline src/sage/rings/integer.pyx]: missing optional

File: src/sage/rings/integer.pyx [hunk 3]
```diff
@@ -3678,44 +3678,44 @@ cdef class Integer(sage.structure.element.EuclideanDomainElement):
...
-            sage: ZZ(0).trial_division()
+            sage: ZZ(0).trial_division()                                            # optional - sage.libs.pari
             Traceback (most recent call last):
             ...
             ValueError: self must be nonzero
 
-            sage: n = next_prime(10^5) * next_prime(10^40); n.trial_division()
+            sage: n = next_prime(10^5) * next_prime(10^40); n.trial_division()      # optional - sage.libs.pari
             100003
             sage: n.trial_division(bound=10^4)
```
Comment (kliem) [inline src/sage/rings/integer.pyx]: Missing optional

File: src/sage/rings/integer.pyx [hunk 4]
```diff
@@ -3678,44 +3678,44 @@ cdef class Integer(sage.structure.element.EuclideanDomainElement):
...
             1000030000000000000000000000000000000012100363
             sage: (-n).trial_division(bound=10^4)
```
Comment (kliem) [inline src/sage/rings/integer.pyx]: Missing optional

File: src/sage/rings/integer.pyx [hunk 5]
```diff
@@ -3678,44 +3678,44 @@ cdef class Integer(sage.structure.element.EuclideanDomainElement):
...
             1000030000000000000000000000000000000012100363
             sage: (-n).trial_division()
```
Comment (kliem) [inline src/sage/rings/integer.pyx]: Missing optional

**Review by kliem — CHANGES_REQUESTED** [review]
There are some failing tests, which makes me wonder whether `# optional - sage.libs.pari` is working at all.

**Comment by mkoeppe** [comment]
Indeed the optional tag was not working, fixed now.

**Review by kliem — APPROVED** [review]

**Comment by mkoeppe** [comment]
Thank you!

**Review by kliem — APPROVED** [review]

**Pull request merged**